Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/random_perturbation.h

```diff
@@ -0,0 +1,172 @@
+//
+// $URL$
+// $Id$
+//
+//
+// Author(s)     : Jane Tournois
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_RANDOM_PERTURBATION_H
+#define CGAL_POLYGON_MESH_PROCESSING_RANDOM_PERTURBATION_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+#include <CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h>
+
+#include <CGAL/Polygon_mesh_processing/internal/named_function_params.h>
+#include <CGAL/Polygon_mesh_processing/internal/named_params_helper.h>
+
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_traits.h>
+#include <CGAL/AABB_face_graph_triangle_primitive.h>
+
+#include <CGAL/Random.h>
+
+#include <boost/foreach.hpp>
+
+#ifdef CGAL_PMP_RANDOM_PERTURBATION_VERBOSE
+#include <CGAL/Timer.h>
+#endif
+
+namespace CGAL {
+
+namespace Polygon_mesh_processing {
+
+namespace internal {
+
+  template<typename GT, typename PM, typename VCMap, typename VPMap>
+  void random_perturbation_impl(PM& tmesh,
+                                const double& max_size,
+                                VCMap vcmap,
+                                VPMap vpmap,
+                                const int seed)
+  {
+    typedef typename boost::graph_traits<PM>::vertex_descriptor vertex_descriptor;
+    typedef typename GT::Point_3    Point_3;
+    typedef typename GT::FT         FT;
+
+    typedef CGAL::AABB_face_graph_triangle_primitive<PM> Primitive;
+    typedef CGAL::AABB_traits<GT, Primitive> Traits;
+    typedef CGAL::AABB_tree<Traits> Tree;
+
+    Tree tree(faces(tmesh).first, faces(tmesh).second, tmesh);
+    tree.accelerate_distance_queries();
+
+    CGAL::Random rng(seed);
+
+    BOOST_FOREACH(vertex_descriptor v, vertices(tmesh))
+    {
+      if (!get(vcmap, v) && !is_border(v, tmesh))
+      {
+        const Point_3& p = get(vpmap, v);
+        const Point_3 np(p.x() + FT(rng.get_double(-max_size, max_size)),
+                         p.y() + FT(rng.get_double(-max_size, max_size)),
+                         p.z() + FT(rng.get_double(-max_size, max_size)));
+        const Point_3 closest = tree.closest_point(np); //project on input surface
+
+        put(vpmap, v, closest);
+      }
+    }
+  }
+
+} //end namespace internal
+
+/*!
+* \ingroup PMP_meshing_grp
+* @brief randomly perturbs the locations of vertices of a triangulated surface mesh.
+* Note that depending on the chosen parameters, inversions and self-intersections may happen.
+*
+* @tparam TriangleMesh model of `MutableFaceGraph`.
+* @tparam NamedParameters a sequence of \ref namedparameters
+*
+* @param tmesh the triangulated surface mesh to be perturbed
+* @param perturbation_max_size the maximal length of moves that can be applied to
+*        vertices of `tmesh`.
+* @param np optional sequence of \ref namedparameters among the ones listed below
+*
+* \cgalNamedParamsBegin
+*  \cgalParamBegin{geom_traits} a geometric traits class instance, model of `Kernel`.
+*    Exact constructions kernels are not supported by this function.
```

Comment:
I think in "Exact constructions kernels" exact constructions is used as an adjective and it should be singular. A better formulation would be "Kernels with exact constructions ..."